src/test/java/org/opensearch/ad/bwc/ADBackwardsCompatibilityIT.java

```diff
@@ -83,25 +131,176 @@ public void testBackwardsCompatibility() throws Exception {
+                AnomalyDetectorJob job = (AnomalyDetectorJob) jobAndTask.get(ANOMALY_DETECTOR_JOB);
+                ADTask historicalTask = (ADTask) jobAndTask.get(HISTORICAL_ANALYSIS_TASK);
+                ADTask realtimeTask = (ADTask) jobAndTask.get(REALTIME_TASK);
+                assertTrue(job.isEnabled());
+                assertNotNull(historicalTask);
+                if (realtimeTask == null) {
+                    realtimeTaskMissing = true;
+                }
+                if (i >= maxRetryTimes) {
+                    assertNotNull(realtimeTask);
+                    assertFalse(realtimeTask.isDone());
+                }
+            }
+            if (realtimeTaskMissing) {
+                Thread.sleep(10_000);// sleep 10 seconds to wait for realtime job to backfill realtime task
+            }
+        } while (realtimeTaskMissing && i < maxRetryTimes);
+    }
+
+    private void stopAndDeleteDetectors() throws Exception {
+        for (String detectorId : runningRealtimeDetectors) {
+            deleteRunningDetector(detectorId);
+            Response stopRealtimeJobResponse = stopRealtimeJob(client(), detectorId);
+            assertEquals(RestStatus.OK, TestHelpers.restStatus(stopRealtimeJobResponse));
+            try {
+                Response stopHistoricalAnalysisResponse = stopHistoricalAnalysis(client(), detectorId);
+                assertEquals(RestStatus.OK, TestHelpers.restStatus(stopHistoricalAnalysisResponse));
+            } catch (Exception e) {
+                if (!ExceptionUtil.getErrorMessage(e).contains("No running task found")) {
+                    throw e;
+                }
+            }
+            Map<String, Object> jobAndTask = getDetectorWithJobAndTask(client(), detectorId);
+            AnomalyDetectorJob job = (AnomalyDetectorJob) jobAndTask.get(ANOMALY_DETECTOR_JOB);
+            ADTask historicalAdTask = (ADTask) jobAndTask.get(HISTORICAL_ANALYSIS_TASK);
+            if (!job.isEnabled() && historicalAdTask.isDone()) {
+                Response deleteDetectorResponse = deleteDetector(client(), detectorId);
+                assertEquals(RestStatus.OK, TestHelpers.restStatus(deleteDetectorResponse));
+            }
+        }
+    }
+
+    private void startHistoricalAnalysisOnNewNode(String detectorId, String taskType) throws IOException, InterruptedException {
+        String taskId = startHistoricalAnalysis(client(), detectorId);
+        deleteRunningDetector(detectorId);
+        waitUntilTaskDone(client(), detectorId);
+        List<ADTask> adTasks = searchLatestAdTaskOfDetector(client(), detectorId, taskType);
+        assertEquals(1, adTasks.size());
+        assertEquals(taskId, adTasks.get(0).getTaskId());
+        int adResultCount = countADResultOfDetector(client(), detectorId, taskId);
+        assertTrue(adResultCount > 0);
+    }
+
+    private void startRealtimeJobForHistoricalDetectorOnNewNode() throws IOException {
```

Comment:
these 2 methods include `OnNewNode` when that functionality hasn't been added yet. Maybe rename them, or add comments above them saying that these can't guarantee they are running on new nodes yet until that functionality's available?

Reply:
"OnNewNode" means we can only execute these functions on new AD node, if these functions executed on old nodes, they will generate unexpected result.

These functions will run in `CASE UPGRADED` branch, which means fully upgraded cluster, all nodes in this cluster are running new AD version. If BWC test framework supports sending request to a specific node, we can run these functions on only new nodes in mixed cluster as well. But that's impossible to send request to a specific node via REST API now. That will be future work.
```
case "upgraded_cluster":
                    return UPGRADED;
```

Reply:
Got it, makes sense. Could you add comments explaining that? So it will be easy to integrate these when the mixed-cluster-specific-node-version functionality is supported?

Reply:
sure